### ISSUE TITLE
[BUG]: Cannot disable squash merge because of 422 Validation Failed 

### ISSUE THREAD

**Issue by ecolomer** [integrations/terraform-provider-github#2147]
### Expected Behavior

When I run a `terraform plan` for `github_repository` resources, which have `allow_squash_merge` set to `false`, I should not see any values for `squash_merge_commit_message` and `squash_merge_commit_title`. 

### Actual Behavior

The provider [code](https://github.com/integrations/terraform-provider-github/blob/main/github/resource_github_repository.go#L180) sets a default for `squash_merge_commit_message` and `squash_merge_commit_title`, even when `allow_squash_merge` is set to `false`.

And then, when I run a `terraform apply`, I get the following error:
```
Error: PATCH https://api.github.com/repos/xxxx/xxxx: 422 Validation Failed [{Resource:Repository Field:merge_commit_allowed Code:invalid Message:Sorry, you need to allow the squash merge strategy in order to set the default squash commit message title or message. (no_squash_merge_strategy)}]
```

### Terraform Version

Terraform v1.2.9 on darwin_arm64 + provider registry.terraform.io/hashicorp/tls v4.0.5 + provider registry.terraform.io/integrations/github v5.45.0

### Affected Resource(s)

- github_repository

### Terraform Configuration Files

```hcl
resource "github_repository" "default" {
  allow_auto_merge            = false
  allow_merge_commit          = true
  allow_rebase_merge          = false
  allow_squash_merge          = false
  archive_on_destroy          = true
  archived                    = false
  auto_init                   = true
  delete_branch_on_merge      = false
  has_downloads               = false
  has_issues                  = false
  has_projects                = false
  has_wiki                    = false
  is_template                 = false
  name                        = "my-test-repo"
  visibility                  = "internal"
}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by JackPGreen** [comment]
[Related](https://github.com/PyGithub/PyGithub/issues/2890#issuecomment-2104361895).

**Comment by rwblokzijl** [comment]
This is still not fixed. These errors keep showing up when updating the repo, even when the creation worked fine.

**Comment by jbscare** [comment]
Agreed, this is still not fixed, please reopen this issue and fix it? :^ )

**Comment by deiga** [comment]
@jbscare could you please open a New issue and fill out all the details to help us debug this?
